### ISSUE TITLE
Validation failed: Scopes doesn't match configured on the server

### ISSUE THREAD

**Issue by Lohn** [snarfed/bridgy#1044]
Hi! I'm trying to log into my personal mastodon instance social.lohn.in. 
But when I click to login I get this message.
HTTP Error 422: {"error":"Validation failed: Scopes doesn't match configured on the server."}

It's running the last version 3.4.1.
I tried in another instance not controlled by me social.tchncs.de which runs the same version and everything runs OK.
So I think the problem is in my instance. But the only thing different on that is that the web domain social.lohn.in differs from the instance which is only lohn.in. Mastodon permits it if we redirect the .well-known URLS. This is the first problem I get, but I don't know id it's related.

I know mastodon is hard to debug so I got the requisitions from my proxy:

1st Req (X - Origin, Y - Destiny)
{"srcIP":"XXX.XXX.XXX.XXX","backend":"social","bIP":"YYY.YYY.YYY.YYY","bPORT":"3000","method":"GET","user-agent":"python-requests\/2.25.1","uri":"/api/v1/instance","body":"-"}

2nd Rex (X - Origin, Y - Destiny)
{"srcIP":"XXX.XXX.XXX.XXX","backend":"social","bIP":"YYY.YYY.YYY.YYY","bPORT":"3000","method":"POST","user-agent":"python-requests\/2.25.1","uri":"/api/v1/apps","body":"client_name=Bridgy&redirect_uris=https%3A%2F%2Fbrid.gy%2Fmastodon%2Fdelete%2Ffinish%0Ahttps%3A%2F%2Fbrid.gy%2Fpublish%2Fmastodon%2Ffinish%0Ahttps%3A%2F%2Fbrid.gy%2Fmastodon%2Fcallback%0Ahttps%3A%2F%2Fbrid.gy%2Fdelete%2Ffinish&website=https%3A%2F%2Fbrid.gy&scopes=read+read%3Aaccounts+read%3Ablocks+read%3Afavourites+read%3Afilters+read%3Afollows+read%3Alists+read%3Amutes+read%3Anotifications+read%3Areports+read%3Asearch+read%3Astatuses+write+write%3Aaccounts+write%3Ablocks+write%3Afavourites+write%3Afilters+write%3Afollows+write%3Alists+write%3Amedia+write%3Amutes+write%3Anotifications+write%3Areports+write%3Astatuses+follow+push"}

If I could say, the problem is those big scopes that bridgy is asking. And read:reports is deprecated (probably the one giving the scope error)

Anyway, if you can enlighten the problem, I can make any test/change you ask.

**Comment by snarfed** [comment]
Hey, thanks for filing, and for the detailed info! I'll take a look.

**Comment by snarfed** [comment]
@Lohn could you try again? I haven't change anything on Bridgy's end, but I tried to reproduce just now, and it worked ok. The `/api/v1/apps` API call that failed last time succeeded this time, so Bridgy was able to create an API client on your instance, and it then correctly redirected me to https://social.lohn.in/auth/sign_in.

Maybe something changed on your end? Or it was transient somehow?

**Comment by snarfed** [comment]
Actually, sorry, I take it back, I may have changed something on Bridgy's end after all. I'll check in the morning.

**Comment by snarfed** [comment]
Confirmed. [Mastodon deprecated the `read:reports` scope back in 2.6.0](https://docs.joinmastodon.org/api/oauth-scopes/#versions), which Bridgy was still requesting. There's [no mention in the Mastodon release notes](https://github.com/tootsuite/mastodon/releases) of actually removing that scope, and it's still really odd and suspicious that other 3.4.1 instances like social.tchncs.de still accept it ok.

Regardless, I removed it from Bridgy's API call, which is probably what fixed this for social.lohn.in.
